R/runCheck.R

```diff
@@ -38,31 +39,50 @@ runCheck <- function(checkXML, metadataXML, sysmetaXML, checkFunction) {
     sysmeta <- sysmetaXML
   }
 
-  # Read in the metadata document that will be checked.
-  metadataDoc <- read_xml(metadataXML)
-  # Create a copy of the document that is not namespace aware (i.e. namespace definitions stripped).
-  # This document will be used for selectors that don't have namespaces defined, and therefore use
-  # local XML paths.
-  metadataDocNoNS <- read_xml(metadataXML)
-  # nsList contains a list of namespace prefixes and associated URIs
-  nsList <- xml_ns(metadataDoc)
-  nsPrefixes <- names(nsList)
-  if (length(nsPrefixes) > 0) {
-    for (thisNsPrefix in nsPrefixes) {
-      # Don't remove the XMLSchema-instance namespace
-      if (thisNsPrefix == "xsi") next
-      metadataDocNoNS <- ns_strip(metadataDocNoNS, thisNsPrefix)
+  # Read either JSON -or- XML
+  metadata_file_ext <- tolower(tools::file_ext(metadataFILE))
+  isXML <- metadata_file_ext == 'xml'
+  isSchemaOrg <- FALSE # This will be changed to TRUE in second if chain
+  if(isXML) {
+    # Read in the metadata document that will be checked.
+    metadataDoc <- read_xml(metadataFILE)
+    # Create a copy of the document that is not namespace aware (i.e. namespace definitions stripped).
+    # This document will be used for selectors that don't have namespaces defined, and therefore use
+    # local XML paths.
+    metadataDocNoNS <- read_xml(metadataFILE)
+    # nsList contains a list of namespace prefixes and associated URIs
+    nsList <- xml_ns(metadataDoc)
+    nsPrefixes <- names(nsList)
+    if (length(nsPrefixes) > 0) {
+      for (thisNsPrefix in nsPrefixes) {
+        # Don't remove the XMLSchema-instance namespace
+        if (thisNsPrefix == "xsi") next
+        metadataDocNoNS <- ns_strip(metadataDocNoNS, thisNsPrefix)
+      }
     }
+  } else if(!isXML & metadata_file_ext == 'json') {
+    # Handle JSON
+    # Read in the metadata document that will be checked.
+    metadataDoc <- paste(readLines(metadataFILE), collapse=' ') # `jq` functions need JSON as string
+    metadataDocNoNS <- metadataDoc # Namespaces don't matter for JSON, so just make a copy
+    # Currently only supporting schema.org
+    if(!grepl('schema.org', metadataDoc))
```

Comment:
it would probably be more robust to run an actual jq expression here, checking that `@context` is `https://schema.org` as opposed to looking in the entire document for mention of schema.org

Reply:
yes, for sure. Also note that we already have a defined check for this in our SHACL shapes. It is quite tricky to properly interpret the schema context URIs. There's a long discussion of this in the SOSO github issues and guidance document.